docs/docs/contribute/software/add-new-metric-provider.md

```diff
@@ -0,0 +1,107 @@
+
+2. **Define the Provider Type:** In the `metrics-operator/controllers/common/providers/common.go` file,
+ define the constant `KeptnPlaceholderProviderType`.
+  In our example we use `"placeholder"`.
+
+    ```go
+    const KeptnPlaceholderProviderType = "placeholder"
+    ```
+
+3. **Implement the Provider:** Create your own new folder inside the
+[metrics-operator/controllers/common/providers](https://github.com/keptn/lifecycle-toolkit/tree/main/metrics-operator/controllers/common/providers),
+ matching the new provider name
+ (`placeholder` in our example).
+Create a new Go package for the placeholder provider in that folder.
+  This package should contain
+ a `struct` that implements the `KeptnSLIProvider` interface.
+  To fully implement the `KeptnSLIProvider` interface, it's necessary to implement the following functions.
+  `EvaluateQuery`(Fetches metric values from the provider)
+   - This function fetches metric values based on the provided
+     metric query from the provider.
+     It evaluates the query and returns the metric values
+     along with any additional data if required.
+   - It takes as input a [KeptnMetric]
+   (<https://github.com/keptn/lifecycle-toolkit/blob/main/metrics-operator/api/v1beta1/keptnmetric_types.go>)
+   and [KeptnMetricsProvider](https://github.com/keptn/lifecycle-toolkit/blob/main/metrics-operator/api/v1beta1/keptnmetricsprovider_types.go)
+  `EvaluateQueryForStep`(Fetches metric values with step interval from the provider)
+   - This function fetches metric values with a specified step interval from the placeholder provider.
+      It takes into account the metric query and the step interval provided, executes the query,
+      and returns the metric values along with any additional data if required.
+   - It takes as input a [KeptnMetric]
+   (<https://github.com/keptn/lifecycle-toolkit/blob/main/metrics-operator/api/v1beta1/keptnmetric_types.go>)
+   and [KeptnMetricsProvider](https://github.com/keptn/lifecycle-toolkit/blob/main/metrics-operator/api/v1beta1/keptnmetricsprovider_types.go)
+  `FetchAnalysisValue`(Fetches analysis values from the provider) functions.
+   - This function fetches analysis values based on the provided query and time range from the
+     provider.
+     It evaluates the query within the specified time range and returns the analysis
+     values along with any additional data if required.
+   - It takes as input a `query`, [Analysis]
+   (<https://github.com/keptn/lifecycle-toolkit/blob/main/metrics-operator/api/v1beta1/analysis_types.go>) and [KeptnMetricsProvider](https://github.com/keptn/lifecycle-toolkit/blob/main/metrics-operator/api/v1beta1/keptnmetricsprovider_types.go)
+  You can follow other existing implementations,
+ such as [prometheus.go](https://github.com/keptn/lifecycle-toolkit/blob/main/metrics-operator/controllers/common/providers/prometheus/prometheus.go),
+ as an example.
+   **NB:** All 3 functions expects a string containing a float value in it.
```

Comment:
```suggestion
   **NB:** Each of the three functions expects a string containing a float value in it.
```
Or you could say "All of the three functions expect..." but "Each" is probably better.